codec-dns/src/main/java/io/netty/handler/codec/dns/AbstractDnsOptPseudoRrRecord.java

```diff
@@ -38,7 +38,7 @@ protected AbstractDnsOptPseudoRrRecord(int maxPayloadSize) {
     // See https://tools.ietf.org/html/rfc6891#section-6.1.3
     private static long packIntoLong(int val, int val2) {
         // We are currently not support DO and Z fields, just use 0.
-        return ((val & 0xff) << 24 | (val2 & 0xff) << 16 | (0 & 0xff) <<  8 | 0 & 0xff) & 0xFFFFFFFFL;
+        return ((long) (val & 0xff) << 24 | (val2 & 0xff) << 16) & 0xFFFFFFFFL;
```

Comment:
@chrisvest Can you double-check this? Whether it's correct or not?

Reply:
Looks correct. You can use `0xFFL` as mask, and then you won't need the explicit long-cast.

Reply:
Instead of `(long) (val & 0xff) << 24` you can do `(val & 0xffL) << 24`.